src/Aspire.Dashboard/Components/Controls/ResourceDetails.razor.cs

```diff
@@ -0,0 +1,226 @@
+    private readonly Icon _showAllIcon = new Icons.Regular.Size16.DocumentOnePage();
+
+    private readonly GridSort<EnvironmentVariableViewModel> _nameSort = GridSort<EnvironmentVariableViewModel>.ByAscending(vm => vm.Name);
+    private readonly GridSort<EnvironmentVariableViewModel> _valueSort = GridSort<EnvironmentVariableViewModel>.ByAscending(vm => vm.Value);
+
+    private List<KnownProperty> _resourceProperties = default!;
+    private List<KnownProperty> _projectProperties = default!;
+    private List<KnownProperty> _executableProperties = default!;
+    private List<KnownProperty> _containerProperties = default!;
+
+    protected override void OnInitialized()
+    {
+        // Known properties can't be static because they need the localizer for translations.
+        _resourceProperties =
+        [
+            new KnownProperty(KnownProperties.Resource.DisplayName, Loc[Resources.Resources.ResourcesDetailsDisplayNameProperty]),
+            new KnownProperty(KnownProperties.Resource.State, Loc[Resources.Resources.ResourcesDetailsStateProperty]),
+            new KnownProperty(KnownProperties.Resource.CreateTime, Loc[Resources.Resources.ResourcesDetailsStartTimeProperty])
+        ];
+        _projectProperties =
+        [
+            .. _resourceProperties,
+            new KnownProperty(KnownProperties.Project.Path, Loc[Resources.Resources.ResourcesDetailsProjectPathProperty]),
+            new KnownProperty(KnownProperties.Executable.Pid, Loc[Resources.Resources.ResourcesDetailsExecutableProcessIdProperty]),
+        ];
+        _executableProperties =
+        [
+            .. _resourceProperties,
+            new KnownProperty(KnownProperties.Executable.Path, Loc[Resources.Resources.ResourcesDetailsExecutablePathProperty]),
+            new KnownProperty(KnownProperties.Executable.WorkDir, Loc[Resources.Resources.ResourcesDetailsExecutableWorkingDirectoryProperty]),
+            new KnownProperty(KnownProperties.Executable.Args, Loc[Resources.Resources.ResourcesDetailsExecutableArgumentsProperty]),
+            new KnownProperty(KnownProperties.Executable.Pid, Loc[Resources.Resources.ResourcesDetailsExecutableProcessIdProperty]),
+        ];
+        _containerProperties =
+        [
+            .. _resourceProperties,
+            new KnownProperty(KnownProperties.Container.Image, Loc[Resources.Resources.ResourcesDetailsContainerImageProperty]),
+            new KnownProperty(KnownProperties.Container.Id, Loc[Resources.Resources.ResourcesDetailsContainerIdProperty]),
+            new KnownProperty(KnownProperties.Container.Command, Loc[Resources.Resources.ResourcesDetailsContainerCommandProperty]),
+            new KnownProperty(KnownProperties.Container.Args, Loc[Resources.Resources.ResourcesDetailsContainerArgumentsProperty]),
+            new KnownProperty(KnownProperties.Container.Ports, Loc[Resources.Resources.ResourcesDetailsContainerPortsProperty]),
+        ];
```

Comment:
We shouldn't need to hard-code known properties for each resource type here. The `ResourceProperty` message received via gRPC includes a display name field. The dashboard should avoid special knowledge about specific fields wherever possible.

It also means we don't have to keep this code up to date as properties are added/removed.

Reply:
We don't **_need_** to, but it improves the experience. They're a balance between making all data for a resource available, while still having a good default experience.

Known properties server three purposes:
1. By default, display a curated list of information that we think is important. For example, it's useful to know the state, while resource.uid is an internal value.
2. Controlling order of information. Ensures the most useful information is first.
3. Displaying a friendly text in the name column.

This is what it looks like without them:
![image](https://github.com/dotnet/aspire/assets/303201/6b1b10e1-9fc4-4d47-b637-feca20e4bcb8)

Updating known properties is optional. Data is still available in the UI if not known. Updating UI to show new data is a normal part of app development.

Reply:
You have to move the loc strings to the server side. All reference data that governs the UI should come from the back end so that we don't have to add explicit support for custom resources in the dashboard.

Reply:
https://github.com/dotnet/aspire/issues/1644